pkg/steps/git_source.go

```diff
@@ -0,0 +1,65 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+
+	buildapi "github.com/openshift/api/build/v1"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+
+	"github.com/openshift/ci-operator/pkg/api"
+)
+
+type gitSourceStep struct {
+	config      api.ProjectDirectoryImageBuildStepConfiguration
+	resources   api.ResourceConfiguration
+	imageClient imageclientset.ImageV1Interface
+	buildClient BuildClient
+	jobSpec     *api.JobSpec
+}
+
+func (s *gitSourceStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return s.jobSpec.Inputs(), nil
+}
+
+func (s *gitSourceStep) Run(ctx context.Context, dry bool) error {
+	return handleBuild(s.buildClient, buildFromSource(s.jobSpec, "", s.config.To, buildapi.BuildSource{
+		Type:       buildapi.BuildSourceGit,
+		ContextDir: s.config.ContextDir,
+		Git: &buildapi.GitBuildSource{
+			URI: fmt.Sprintf("https://github.com/%s/%s.git", s.jobSpec.Refs.Org, s.jobSpec.Refs.Repo),
```

Comment:
How will we handle private repositories in the future?

Reply:
add Uri and Ref fields in the test config?